src/algebra/module/basic.lean

```diff
@@ -526,11 +526,8 @@ include R
 lemma nat.no_zero_smul_divisors : no_zero_smul_divisors ℕ M :=
 ⟨by { intros c x, rw [nsmul_eq_smul_cast R, smul_eq_zero], simp }⟩
 
-variables {M}
-
-lemma eq_zero_of_two_nsmul_eq_zero {v : M} (hv : 2 • v = 0) : v = 0 :=
-by haveI := nat.no_zero_smul_divisors R M;
-exact (smul_eq_zero.mp hv).resolve_left (by norm_num)
+@[simp] lemma two_nsmul_eq_zero {v : M} : 2 • v = 0 ↔ v = 0 :=
```

Comment:
Looking at the others lemma `two_nsmul_eq_zero_iff_eq_zero` seems a better name.

Reply:
I would also be happy with dropping the `_eq_zero` from the other names.

Reply:
As far as I can see there is only a unique way to finish `two_nsmul_eq_zero[_iff]` into something sensible. This is not true for the statements, since it's not clear that the LHS has the same variable twice. I could rename the others to `eq_neg_self[_iff]` if you think that is better.